api/server/httputils/errors.go

```diff
@@ -37,49 +33,45 @@ func GetHTTPErrorStatusCode(err error) int {
 	}
 
 	var statusCode int
-	errMsg := err.Error()
 
-	switch e := err.(type) {
-	case httpStatusError:
-		statusCode = e.HTTPErrorStatusCode()
-	case inputValidationError:
+	// Stop right there
+	// Are you sure you should be adding a new error class here? Do one of the existing ones work?
+
+	// Note that the below functions are already checking the error causal chain for matches.
+	switch {
```

Comment:
> It seems like we still have this problem (from #32144):

I do not agree with this statement. We have pieces of code communicating the class of error that occurred, e.g... "I couldn't find the thing", "You requested something I can't do", "This configuration doesn't make sense", etc.

What's really nice about this is the low-level piece can communicate what it knows the error is, and the higher levels can wrap the error in a different error class if needed, or it can just wrap the error with more context and the correct error class is found using the causal interface.

> A 1:1 mapping from "package error" to "http error"

There is not a 1:1 mapping here. There is certainly a strong correlation of the kinds of errors that can be communicated with http status codes, but this is because that's what the status codes are designed to encapsulate.